### PR TITLE
Support for gradient accumulation and mixed precision training

### PR DESCRIPTION
**Pull request by NaleRaphael**
Hi, @davidtvs , this feature request is mentioned in issue #8 .
User now can use `AccumualtionLRFinder` to find learning rate with mechanism of gradient accumulation.

And I've done some tests to compare the difference of finding learning rate with/without gradient accumulation (in `examples/lrfinder_mnist_gradient_accumulation.ipynb`), it seems that gradient accumulation works fine with `LRFinder`.

If there is anything need to be improved, please feel free to let me know.

### CODE REVIEW

File: torch_lr_finder/lr_finder.py [hunk 1]
```diff
@@ -338,3 +414,30 @@ def __del__(self):
...
+        except StopIteration:
+            if not self.auto_reset:
+                raise
+            self._iterator = iter(self.data_loader)
+            inputs, labels = next(self._iterator)
+
+        return inputs, labels
+
+    def get_batch(self):
+        return next(self)
```
Comment (davidtvs) [inline torch_lr_finder/lr_finder.py]: For Python 2 support add the `next` function:
```
# For Python 2
next = __next__
```

File: README.md [hunk 2]
```diff
@@ -16,6 +16,9 @@ For cyclical learning rates (also detailed in Leslie Smith's paper) where the le
 Python 2.7 and above:
 ``pip install torch-lr-finder``
 
+Install with the support of mixed precision training:
+``pip install torch-lr-finder -v --global-option="amp"``
+
```
Comment (davidtvs) [inline README.md]: Apex requires Python 3, so we should state that, or tell the reader to check Apex requirements and link to the repo

File: torch_lr_finder/lr_finder.py [hunk 3]
```diff
@@ -6,6 +6,16 @@
 from torch.optim.lr_scheduler import _LRScheduler
 import matplotlib.pyplot as plt
 
+try:
+    from apex import amp
+    IS_AMP_AVAILABLE = True
+except ImportError:
+    import logging
+    logger = logging.getLogger(__name__)
```
Comment (davidtvs) [inline torch_lr_finder/lr_finder.py]: Is there a specific use-case for using `logging` here?

Also, `logging.getLogger(__name__)` fails on Python 2 and outputs: `No handlers could be found for logger "torch_lr_finder.lr_finder"`

File: torch_lr_finder/lr_finder.py [hunk 4]
```diff
@@ -6,6 +6,16 @@
...
+    logger.warning('To enable mixed precision training, please install `apex`.')
```
Comment (davidtvs) [inline torch_lr_finder/lr_finder.py]: I would add the link to the apex repository or the command to reinstall this package with the amp option: `pip install torch-lr-finder -v --global-option="amp"`

File: torch_lr_finder/lr_finder.py (hunk 1 above)
Comment (NaleRaphael) [reply torch_lr_finder/lr_finder.py]: Got it! I did forget to test it on Python 2, my bad.

File: README.md (hunk 2 above)
Comment (NaleRaphael) [reply README.md]: I'm with you on that. And I'll add a simple check in `setup.py` to check that, too.

File: torch_lr_finder/lr_finder.py (hunk 3 above)
Comment (NaleRaphael) [reply torch_lr_finder/lr_finder.py]: In my opinion, using a logger to show that message makes users able to silence it once they want to do so. And user can configure the logger before `import torch_lr_finder`, like the code as follows:
```python
# set logging level to ERROR to silence the warning message
import logging
logging.getLogger("torch_lr_finder.lr_finder").setLevel(logging.ERROR)

# warning message "To enable mixed precision training..." won't be shown if apex is not installed
from torch_lr_finder import LRFinder
```

As for the message `No handlers could be found for logger "torch_lr_finder.lr_finder"`, it seems like we have to manually call `logging.basicConfig()` before `logging.getLogger()` in Python 2. I'll add it into the code.

File: torch_lr_finder/lr_finder.py (hunk 4 above)
Comment (NaleRaphael) [reply torch_lr_finder/lr_finder.py]: Ok, I'll add the installation command in the message!

**Comment by davidtvs** [comment]
LGTM. Thanks for the PR, well done @NaleRaphael

**Comment by NaleRaphael** [comment]
It's also my pleasure to work with you! :)
